framework/src/systems/NonlinearEigenSystem.C

```diff
@@ -140,10 +140,12 @@ void
 NonlinearEigenSystem::initialSetup()
 {
   NonlinearSystemBase::initialSetup();
-
+  // Regular kernels
   addEigenTagToMooseObjects(_kernels);
-
+  // Nodal BCs (we do not care about IBCs)
```

Comment:
Why don't we care?

Reply:
By the definition of eigenvalue problems, we require the integral BCs equal zero. And then the integral BCs won't do anything at all in the eigenvalue problem.

Reply:
@fdkong This is not true. I guess you have homogeneous Neumann BC in mind, which is indeed naturally imposed for diffusion problem. Eigenvalue problems should allow us to use something like homogeneous Robin BC. This issue was brought up by @loganharbour yesterday. I think we should have a virtual function `bool BoundaryCondition::isHomogeneous() const`, that can be used for integrity check in eigen problems.

Reply:
I do not think I understand Robin BCs for eigenvalue problems. 

Could you make an example?

Reply:
https://github.com/idaholab/moose/blob/next/framework/src/bcs/VacuumBC.C is actually the Robin BC.

Reply:
Got it.